docker-app/worker_wrapper/wrapper.py

```diff
@@ -226,6 +227,9 @@ def _run_docker(
             detach=True,
         )
 
+        self.job.container_id = container.id
+        self.job.save()
+
```

Comment:
Potentially more trouble some race condition here, as some time could elapse between the job creation above (without container id) and the starting of the container. In the meantime the cronjob could delete the job ?

Maybe better to not create the job until you have a container ID (e.g. move the lines 99:101 to here). You can create directly a failed job if it doesn't complete.

Reply:
Good note, this is very valid situation.